src/Strings.h

```diff
@@ -178,6 +142,19 @@ void ParseNgramCharwbCopy(std::string const &input,
 // ----------------------------------------------------------------------
 namespace Details {
 
+template <typename UnaryPredicateT>
+std::string StringPadding(std::string input, UnaryPredicateT isPredicate) {
+    if (!isPredicate(input.at(0)) && !isPredicate(input.at(input.length() - 1))) {
```

Comment:
I'm guessing that these predicates are not expensive to call. However, they may be. Consider assigning both results to a bool value and then using those values in the conditions so that they are only invoked once.